include/aws/http/server.h

```diff
@@ -103,10 +103,14 @@ struct aws_http_server_options {
 #define AWS_HTTP_SERVER_OPTIONS_INIT                                                                                   \
     { .self_size = sizeof(struct aws_http_server_options), .initial_window_size = SIZE_MAX, }
 
-typedef void(aws_http_on_incoming_request_fn)(
-    struct aws_http_connection *connection,
-    struct aws_http_stream *stream,
-    void *user_data);
+/**
+ * Invoked at the start of an incoming request.
+ * From this callback, the user MUST call aws_http_stream_new_server_request_handler().
+ *
+ * Return AWS_OP_SUCCESS to continue processing the stream.
+ * Return AWS_OP_ERR to indicate failure and cancel the stream.
+ */
+typedef int(aws_http_on_incoming_request_fn)(struct aws_http_connection *connection, void *user_data);
```

Comment:
WHAT IF THIS THING RETURNED THE THING THEY WERE SUPPOSED TO MAKE. Then you could super extra make it clear that this function super double needs to make a request handler.